### DISCUSSION TITLE
Is this package still maintained?

### DISCUSSION THREAD

**Discussion by ChrisTwig** [Opentrons/opentrons-emulation#277] [General]
I want to emulate ot2 to test custom protocols. Is this package still maintained and is it the right way to go?

**Comment by DerekMaggio** [comment]
As of right now, no it is not. 
We are looking to restart this project soon. You should take a look at the simulator over in the opentrons repo. Don't hesitate to reach out to support if you have anything other questions.

https://github.com/Opentrons/opentrons
https://support.opentrons.com/s/article/Simulating-OT-2-protocols-on-your-computer

**Reply by ChrisTwig** [reply]
Thanks for letting me know. I tired via the wiki and it doesn't work as is.

**Reply by koji** [reply]
@ChrisTwig Did you get any errors?

**Reply by ChrisTwig** [reply]
Hey @koji, I managed to get the robot-server started in the end. My changes were:
- convert `monorepo_python` to `python3.10` as the new opentrons package needs it.
- amend the make file to use the commands under entrypoint.sh as there were permission errors with executing the script.

However, I couldn't get it connected through the opentrons app.
I also wasn't too sure what the emulation actually does? I was using OT2 `firmware` emulations.
- Does it store the state? (for instance I did a Post `/robot/lights off` and Get `/robot/lights` and the lights are still `on`.

**Reply by koji** [reply]
when you switch python version to 3.10, did you run `make teardown-py && make setup-py`?


>Does it store the state? (for instance I did a Post /robot/lights off and Get /robot/lights and the lights are still on.
hmm 🤔  I just ran `robot-server` and Robot Light State worked as expected.

https://github.com/Opentrons/opentrons-emulation/assets/474225/d650af47-f9a0-4ecd-9820-7e5551487a58


I think the easiest way to test a protocol with opentrons_simulator is to run the following command.
```shell
opentrons_simulate protocol_name.py
```

You can send your question to the support team.
support@opentrons.com

**Reply by ChrisTwig** [reply]
Interesting. Thanks for the info.
